src/composables/useGetParticipants.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @copyright Copyright (c) 2024 Dorra Jaouad <dorra.jaoued1@gmail.com>
+ *
+ * @author Dorra Jaouad <dorra.jaoued1@gmail.com>
+ * @author Marco Ambrosini <marcoambrosini@icloud.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+import debounce from 'debounce'
+import { ref, onMounted, nextTick, computed } from 'vue'
+
+import { subscribe, unsubscribe } from '@nextcloud/event-bus'
+
+import { useIsInCall } from './useIsInCall.js'
+import { useStore } from './useStore.js'
+import { CONVERSATION } from '../constants.js'
+import { EventBus } from '../services/EventBus.js'
+
+/**
+ * @param {boolean} isActive whether the participants tab is active
+ */
+export function useGetParticipants(isActive = true) {
+
+	// Encapsulation
+	const store = useStore()
+	const token = computed(() => store.getters.getToken())
+	const conversation = computed(() => store.getters.conversation(token.value))
+	const isInCall = useIsInCall()
+	const oneToOneConversation = computed(() => conversation.value.type === CONVERSATION.TYPE.ONE_TO_ONE)
+
+	// External use
+	const participantsInitialised = ref(false)
+	const fetchingParticipants = ref(false)
```

Comment:
```suggestion
	let _fetchingParticipants = false
```
It's internal variable, as I see from the code. Shouldn't be a reactive also

Reply:
Same here

Reply:
There is no need to expose them in composable return and use outside, so they seem private to me. Did the same for another PR just to visually distinct them

Reply:
In JS and Python `_` prefix patter was used where it is not possible to separate private properties and all properties are exposed. So `_` is used to mark exposed property as a private property.

Here variables are private until they are implicitly exposed in `return`. Composable is a function, and variables here are the same local variable as in, for example, services or utils where we also have a local variable and return some of them.

Reply:
Matter of preferences then. I'm not insisting, and will roll it back in my PR for consistency)